src/lib/core/components/visualizations/implementations/LineplotVisualization.tsx

```diff
@@ -1588,10 +1590,18 @@ function nullZeroHack(
   });
 }
 
+/**
+ * Passing the whole of `vizConfig` creates a problem with the TypeScript compiler warnings
+ * for the dependencies of the `data = usePromise(...)` that calls this function. It warns
+ * that `vizConfig` is a missing dependency because it see it being used (passed to `getRequestParams()`)
+ * We can't use the whole of `vizConfig` as a dependency because then data will be re-requested
+ * when only client-side configs are changed. There should probably be two sub-objects of `vizConfig`,
+ * for client and server-side configs.
+ */
 function getRequestParams(
   studyId: string,
   filters: Filter[],
-  vizConfig: LineplotConfig,
+  vizConfig: Omit<LineplotConfig, 'dependentAxisRange' | 'checkedLegendItems'>,
```

Comment:
This was an attempt to solve the `vizConfig` missing dependency problem. Didn't work, but left it in.